### PR TITLE
feat: confirmation message lld

### PR DESCRIPTION
**Pull request by chrisduma-ledger**
### ✅ Checklist

- [x] `npx changeset` was attached.
- [x] **Covered by automatic tests.** 
- [x] **Impact of the changes:** 
  - ...

### 📝 Description

Adds a confirmation message for Sell, after a tx is complete in LLD

### ❓ Context

[Ticket](https://ledgerhq.atlassian.net/browse/LIVE-13293)

---

### 🧐 Checklist for the PR Reviewers

- **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- **Any new dependencies** have been justified and documented.
- **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

### CODE REVIEW

File: apps/ledger-live-desktop/src/renderer/reducers/UI.ts [hunk 1]
```diff
@@ -105,6 +105,7 @@ const handlers: UIHandlers = {
     };
   },
   PLATFORM_APP_DRAWER_OPEN: (state, { payload }) => {
+    console.log("Platform app drawer opened", payload);
```
Comment (CremaFR) [inline apps/ledger-live-desktop/src/renderer/reducers/UI.ts:108]: 🔫

**Review by CremaFR — COMMENTED** [review]
remove console.log no?

**Comment by chrisduma-ledger** [comment]
> remove console.log no?

Yeah. I missed 3, thanks.

**Review by CremaFR — DISMISSED** [review]

**Review by mcayuelas-ledger — APPROVED** [review]
OK for Hub

File: apps/ledger-live-desktop/src/renderer/modals/Platform/Exchange/CompleteExchange/Body.tsx [hunk 2]
```diff
@@ -163,24 +170,30 @@ const Body = ({ data, onClose }: { data: Data; onClose?: () => void | undefined
           return onCancel(new DisabledTransactionBroadcastError());
         }
         onResult(operation);
-        // else not swap i.e card and sell we close the drawer
+      } else if (
+        (data.exchangeType === ExchangeType.SELL || data.exchangeType === ExchangeType.SELL_NG) &&
+        sourceCurrency
+      ) {
```
Comment (sarneijim) [inline apps/ledger-live-desktop/src/renderer/modals/Platform/Exchange/CompleteExchange/Body.tsx:176]: Can we check "quoteId" as I am not sure if an old flow us using "ExchangeType.SELL"

```suggestion
const isValidPtxSellFlow= data.exchangeType === ExchangeType.SELL || data.exchangeType === ExchangeType.SELL_NG) &&
        sourceCurrency
        (isValidPtxSellFlow) {
```

File: apps/ledger-live-desktop/src/renderer/modals/Platform/Exchange/CompleteExchange/Body.tsx [hunk 3]
```diff
@@ -163,24 +170,30 @@ const Body = ({ data, onClose }: { data: Data; onClose?: () => void | undefined
...
+        onResult(operation);
```
Comment (sarneijim) [inline apps/ledger-live-desktop/src/renderer/modals/Platform/Exchange/CompleteExchange/Body.tsx:177]: We need to set DISABLE_TRANSACTION_BROADCAST logic too to avoid to have false errors. (i think we need to use the same flow than swap use: 
setResult
if DISABLE_TRANSACTION_BROADCAST
onResult

```
if (getEnv("DISABLE_TRANSACTION_BROADCAST") 
```

File: apps/ledger-live-desktop/src/renderer/modals/Platform/Exchange/CompleteExchange/Body.tsx [hunk 4]
```diff
@@ -163,24 +170,30 @@ const Body = ({ data, onClose }: { data: Data; onClose?: () => void | undefined
           return onCancel(new DisabledTransactionBroadcastError());
         }
         onResult(operation);
-        // else not swap i.e card and sell we close the drawer
+      } else if (
+        (data.exchangeType === ExchangeType.SELL || data.exchangeType === ExchangeType.SELL_NG) &&
+        sourceCurrency
```
Comment (sarneijim) [inline apps/ledger-live-desktop/src/renderer/modals/Platform/Exchange/CompleteExchange/Body.tsx:175]: Can we check we have sellId too. I do not know if any other old discovery app is using this flow.

File: apps/ledger-live-desktop/src/renderer/modals/Platform/Exchange/CompleteExchange/TransactionBroadcastedContent.tsx [hunk 5]
```diff
@@ -30,24 +32,32 @@ export function TransactionBroadcastedContent(props: TransactionBroadcastedConte
         provider={provider}
         {...swapDefaultTrack}
```
Comment (sarneijim) [inline apps/ledger-live-desktop/src/renderer/modals/Platform/Exchange/CompleteExchange/TransactionBroadcastedContent.tsx:33]: I do not think have sense track swap properties here.

File: apps/ledger-live-desktop/src/renderer/modals/Platform/Exchange/CompleteExchange/TransactionBroadcastedContent.tsx [hunk 6]
```diff
@@ -1,25 +1,27 @@
...
 import SwapCompleted from "~/renderer/screens/exchange/Swap2/Form/ExchangeDrawer/SwapCompleted";
 import { useGetSwapTrackingProperties } from "~/renderer/screens/exchange/Swap2/utils";
 import TrackPage from "~/renderer/analytics/TrackPage";
+import { Currency } from "@ledgerhq/types-cryptoassets";
+import SellCompleted from "~/renderer/screens/exchange/Sell/SellCompleted";
 
 type TransactionBroadcastedContentProps = {
-  swapId: string;
+  swapId?: string;
+  isSell?: boolean;
```
Comment (sarneijim) [inline apps/ledger-live-desktop/src/renderer/modals/Platform/Exchange/CompleteExchange/TransactionBroadcastedContent.tsx:14]: Can we do something more consistent?
```suggestion
  quoteId?: string;
  mode: "sell" | "swap" ;
```

**Review by sarneijim — CHANGES_REQUESTED** [review]
Changes need to be applied

File: apps/ledger-live-desktop/src/renderer/modals/Platform/Exchange/CompleteExchange/Body.tsx (hunk 4 above)
Comment (chrisduma-ledger) [reply apps/ledger-live-desktop/src/renderer/modals/Platform/Exchange/CompleteExchange/Body.tsx:175]: There is no such thing as `sellId`

File: apps/ledger-live-desktop/src/renderer/modals/Platform/Exchange/CompleteExchange/Body.tsx (hunk 3 above)
Comment (chrisduma-ledger) [reply apps/ledger-live-desktop/src/renderer/modals/Platform/Exchange/CompleteExchange/Body.tsx:177]: It is not that important, the flow can go on in this case, but i will add for security.

File: apps/ledger-live-desktop/src/renderer/modals/Platform/Exchange/CompleteExchange/TransactionBroadcastedContent.tsx (hunk 5 above)
Comment (chrisduma-ledger) [reply apps/ledger-live-desktop/src/renderer/modals/Platform/Exchange/CompleteExchange/TransactionBroadcastedContent.tsx:33]: Good catch, will create a hook for sell properties.

File: apps/ledger-live-desktop/src/renderer/modals/Platform/Exchange/CompleteExchange/TransactionBroadcastedContent.tsx (hunk 6 above)
Comment (chrisduma-ledger) [reply apps/ledger-live-desktop/src/renderer/modals/Platform/Exchange/CompleteExchange/TransactionBroadcastedContent.tsx:14]: Since there is no `sellId`, i have added the `isSell` property. This flow is very twisted, having a `mode` would be ideal but I would have a lot to refactor. Don't worry, the logic is pretty sound, the `isSell` will ever be true only if the transaction is a sell.

**Comment by chrisduma-ledger** [comment]
> We could break old UI apps.

Sorry, i do not understand this comment

File: apps/ledger-live-mobile/Gemfile.lock
Comment (Justkant) [inline apps/ledger-live-mobile/Gemfile.lock:1]: You probably want to revert the changes to this file

**Review by Justkant — CHANGES_REQUESTED** [review]

File: apps/ledger-live-mobile/Gemfile.lock
Comment (chrisduma-ledger) [reply apps/ledger-live-mobile/Gemfile.lock:1]: Hello @Justkant, is this not the right way to change the code regarding that package? We need this part that we've added

File: apps/ledger-live-desktop/src/renderer/modals/Platform/Exchange/CompleteExchange/Body.tsx [hunk 7]
```diff
@@ -142,45 +156,62 @@ const Body = ({ data, onClose }: { data: Data; onClose?: () => void | undefined
...
+            result: newResult,
+            magnitudeAwareRate,
+          });
+        }
+
+        const result = isValidExchangeSDKSwapTx
+          ? { swapId, provider, sourceCurrency, targetCurrency, mode: ExchangeModeEnum.Swap }
+          : { provider, mode: ExchangeModeEnum.Sell, sourceCurrency };
+
+        setResult(result);
```
Comment (CremaFR) [inline apps/ledger-live-desktop/src/renderer/modals/Platform/Exchange/CompleteExchange/Body.tsx:190]: Plz simplify with a simple if swap do X else if sell do Y. There is nothing in common in here.

File: libs/ledger-live-common/src/wallet-api/Exchange/server.ts [hunk 8]
```diff
@@ -280,7 +290,7 @@ export const handlers = ({
               binaryPayload: params.hexBinaryPayload,
               exchange,
               feesStrategy: params.feeStrategy,
-              swapId: params.exchangeType === "SWAP" ? params.swapId : undefined,
+              quoteId: quoteOrSwapId,
```
Comment (CremaFR) [inline libs/ledger-live-common/src/wallet-api/Exchange/server.ts:293]: you renamed swapId to quoteId but everywhere else you're still using swapId and you have this todo here
https://github.com/LedgerHQ/ledger-live/pull/7696/files#diff-af77015a1ecc0bc635f1da1c8289441144dd58c96c4535215b6c78920618676fR159

**Review by CremaFR — CHANGES_REQUESTED** [review]

File: apps/ledger-live-desktop/src/renderer/modals/Platform/Exchange/CompleteExchange/Body.tsx (hunk 7 above)
Comment (sarneijim) [inline apps/ledger-live-desktop/src/renderer/modals/Platform/Exchange/CompleteExchange/Body.tsx:190]: We could send data here, and check if it is "swap" later... what do you think?
```suggestion
        setResult(data);
```

File: apps/ledger-live-desktop/src/renderer/modals/Platform/Exchange/CompleteExchange/Body.tsx [hunk 9]
```diff
@@ -142,45 +156,62 @@ const Body = ({ data, onClose }: { data: Data; onClose?: () => void | undefined
...
+        toAccount &&
+        magnitudeAwareRate &&
+        sourceCurrency &&
+        targetCurrency;
+
+      const isValidExchangeSDKSellTx =
+        (data.exchangeType === ExchangeType.SELL || data.exchangeType === ExchangeType.SELL_NG) &&
+        sourceCurrency;
+
+      if (isValidExchangeSDKSellTx || isValidExchangeSDKSwapTx) {
```
Comment (sarneijim) [inline apps/ledger-live-desktop/src/renderer/modals/Platform/Exchange/CompleteExchange/Body.tsx:173]:
```suggestion
      const isValidExchangeSDKTx = quoteId
      if (quoteId) {
```

File: apps/ledger-live-desktop/src/renderer/modals/Platform/Exchange/CompleteExchange/Body.tsx (hunk 7 above)
Comment (CremaFR) [reply apps/ledger-live-desktop/src/renderer/modals/Platform/Exchange/CompleteExchange/Body.tsx:190]: We're checking 3 times isValidExchangeSDKSwapTx, we can only do it once with something like this and without Ifs inside ifs 
```
if (isValidExchangeSDKSwapTx) {
    const newResult = { operation, swapId };
    updateAccount({ result: newResult, magnitudeAwareRate });

    const result = {
        swapId,
        provider,
        sourceCurrency,
        targetCurrency,
        mode: ExchangeModeEnum.Swap,
    };
    setResult(result);
} else if (isValidExchangeSDKSellTx) {
    const result = {
        provider,
        sourceCurrency,
        mode: ExchangeModeEnum.Sell,
    };
    setResult(result);
}```

Comment (sarneijim) [reply apps/ledger-live-desktop/src/renderer/modals/Platform/Exchange/CompleteExchange/Body.tsx:190]: It is valid to me

File: apps/ledger-live-desktop/src/renderer/modals/Platform/Exchange/CompleteExchange/Body.tsx [hunk 10]
```diff
@@ -142,45 +156,62 @@ const Body = ({ data, onClose }: { data: Data; onClose?: () => void | undefined
...
+
+          updateAccount({
+            result: newResult,
+            magnitudeAwareRate,
+          });
+        }
+
+        const result = isValidExchangeSDKSwapTx
+          ? { swapId, provider, sourceCurrency, targetCurrency, mode: ExchangeModeEnum.Swap }
+          : { provider, mode: ExchangeModeEnum.Sell, sourceCurrency };
```
Comment (sarneijim) [inline apps/ledger-live-desktop/src/renderer/modals/Platform/Exchange/CompleteExchange/Body.tsx:188]: Could you please update the code to rewrite swapId to quoteId immediately after calling Ledger Live? Additionally, I think it would be good include the variable mode in the same location.

File: apps/ledger-live-desktop/src/renderer/modals/Platform/Exchange/CompleteExchange/TransactionBroadcastedContent.tsx [hunk 11]
```diff
@@ -28,26 +33,34 @@ export function TransactionBroadcastedContent(props: TransactionBroadcastedConte
         sourceCurrency={sourceCurrency?.name}
         targetCurrency={targetCurrency?.name}
         provider={provider}
-        {...swapDefaultTrack}
+        {...(swapId ? swapDefaultTrack : sellDefaultTrack)}
```
Comment (sarneijim) [inline apps/ledger-live-desktop/src/renderer/modals/Platform/Exchange/CompleteExchange/TransactionBroadcastedContent.tsx:36]: Logic must be base in mode.
```suggestion
{...(mode === 
        mode === ExchangeModeEnum. ? swapDefaultTrack : {flow: mode)}
```

File: apps/ledger-live-desktop/src/renderer/modals/Platform/Exchange/CompleteExchange/Body.tsx (hunk 10 above)
Comment (sarneijim) [reply apps/ledger-live-desktop/src/renderer/modals/Platform/Exchange/CompleteExchange/Body.tsx:188]: If you unify the variable as quoteId, it will serve as the indicator to determine if it is a valid ExchangeSDK transaction.

File: apps/ledger-live-desktop/src/renderer/modals/Platform/Exchange/CompleteExchange/Body.tsx (hunk 7 above)
Comment (sarneijim) [reply apps/ledger-live-desktop/src/renderer/modals/Platform/Exchange/CompleteExchange/Body.tsx:190]: I am not sure if it would be simpler send the complete "data" object here, as we could have different needs in the future.
```
      if (quoteId) {
        if (mode === SWAP {
          const newResult = {
            operation,
            swapId,
          };

          updateAccount({
            result: newResult,
            magnitudeAwareRate,
          });
        }
         setResult(data);
      }
       
```

File: apps/ledger-live-desktop/src/renderer/modals/Platform/Exchange/CompleteExchange/Body.tsx [hunk 12]
```diff
@@ -142,45 +156,62 @@ const Body = ({ data, onClose }: { data: Data; onClose?: () => void | undefined
...
+        setResult(result);
 
         if (getEnv("DISABLE_TRANSACTION_BROADCAST")) {
           return onCancel(new DisabledTransactionBroadcastError());
         }
+
         onResult(operation);
-        // else not swap i.e card and sell we close the drawer
       } else {
+        // else not swap i.e card we close the drawer
```
Comment (sarneijim) [inline apps/ledger-live-desktop/src/renderer/modals/Platform/Exchange/CompleteExchange/Body.tsx:198]: This message is not accurate
// old platform exchange flow

File: apps/ledger-live-desktop/static/i18n/en/app.json [hunk 13]
```diff
@@ -1869,8 +1869,8 @@
       "targetAccount": "Target Account"
     },
     "sell": {
-      "notice": "Verify the Sell details on your device before sending it. The addresses are exchanged securely so you don’t have to verify them.",
-      "confirm": "Confirm Sell transaction"
+      "notice": "Verify the details on your device before sending it. The addresses are exchanged securely so you don’t have to verify them.",
+      "confirm": "Confirm transaction"
```
Comment (sarneijim) [inline apps/ledger-live-desktop/static/i18n/en/app.json:1873]: `src/renderer/components/DeviceAction/rendering.tsx` is using this constants, please create your own constants.

File: apps/ledger-live-mobile/Gemfile.lock [hunk 14]
```diff
@@ -5,11 +5,12 @@ GEM
       base64
       nkf
       rexml
-    activesupport (7.0.8.4)
+    activesupport (6.1.7.8)
```
Comment (sarneijim) [inline apps/ledger-live-mobile/Gemfile.lock:8]: As commented by @Justkant revert this file changes.

File: apps/ledger-live-mobile/Gemfile.lock
Comment (Justkant) [reply apps/ledger-live-mobile/Gemfile.lock:1]: You need to update the ruby dependencies for iOS ?

Comment (chrisduma-ledger) [reply apps/ledger-live-mobile/Gemfile.lock:1]: @Justkant Sorry, i thought you meant the `server` file, no I do not need to, i will revert it!

**Review by CremaFR — APPROVED** [review]
Looking good, thanks 🔥

File: apps/ledger-live-desktop/src/renderer/modals/Platform/Exchange/CompleteExchange/Body.tsx [hunk 15]
```diff
@@ -140,47 +154,87 @@ const Body = ({ data, onClose }: { data: Data; onClose?: () => void | undefined
...
+      if (getEnv("DISABLE_TRANSACTION_BROADCAST")) {
+        return onCancel(new DisabledTransactionBroadcastError());
+      }
+
+      const isSwapTransaction =
+        swapId && toAccount && magnitudeAwareRate && sourceCurrency && targetCurrency;
+
+      const isSellTransaction =
+        (data.exchangeType === ExchangeType.SELL || data.exchangeType === ExchangeType.SELL_NG) &&
+        sourceCurrency;
```
Comment (sarneijim) [inline apps/ledger-live-desktop/src/renderer/modals/Platform/Exchange/CompleteExchange/Body.tsx:210]: I though you want to use:
```suggestion
      const isSellTransaction =
        (data.exchangeType === ExchangeType.SELL || data.exchangeType === ExchangeType.SELL_NG);
```

Comment (chrisduma-ledger) [reply apps/ledger-live-desktop/src/renderer/modals/Platform/Exchange/CompleteExchange/Body.tsx:210]: `sourceCurrency` is just serving as an extra guarding clause, like the swap team does

**Review by sarneijim — APPROVED** [review]

**Review by mcayuelas-ledger — APPROVED** [review]
